### PR TITLE
[SEMVER-MAJOR] Removed unused user properties

### PR DESCRIPTION
**Pull request by davidcheung**
connect to https://github.com/strongloop/loopback/issues/1854

##### Removed properties
- credentials
- challenges
- status
- created
- lastUpdated

### CODE REVIEW

**Comment by davidcheung** [comment]
@slnode test please

File: 3.0-RELEASE-NOTES.md [hunk 1]
```diff
@@ -62,4 +62,11 @@ methods: {
 }
 ```
 
-Please see [related code change](https://github.com/strongloop/loopback/pull/2174) here.
\ No newline at end of file
+Please see [related code change](https://github.com/strongloop/loopback/pull/2174) here.
+
+## Removing unused user properties
```
Comment (bajtos) [inline 3.0-RELEASE-NOTES.md]: We use imperative on L23, pleas do the same for consistency.

```
## Remove unused user properties
```

File: 3.0-RELEASE-NOTES.md [hunk 2]
```diff
@@ -62,4 +62,11 @@ methods: {
...
+- credentials
+- challenges
+- status
+- created
+- lastUpdated
```
Comment (bajtos) [inline 3.0-RELEASE-NOTES.md]: Please add a bit more meat to the release note. Explain how to upgrade apps relying on these properties - users should redefine them in their model JSON files, show an example of such JSON definition.

Also add a link to this pull request, to make it easy to find discussions that lead to the change.

**Comment by davidcheung** [comment]
Updated the description 
@bajtos PTAL

File: 3.0-RELEASE-NOTES.md [hunk 3]
```diff
@@ -62,4 +62,28 @@ methods: {
...
+
+Developers that are relying on these properties, can redefine them in `user.json` or equivalent model.json as follow: 
+```js
+//user.json
+"credentials": {
```
Comment (bajtos) [inline 3.0-RELEASE-NOTES.md]: Please start at the top level of model JSON file to make it more clear where to add this snippet. Also use `json` for highlighting (you can check the source of my comment to see how to do it in markdown):

```json
{
  "name": "MyUser",
  "base": "User",
  "properties": {
    "credentials": { "type": "object" },
    "challenges": { "type": "object" },
    "status": "string",
    "created": "date",
    "lastUpdated": "date"
  }
} 
```

File: 3.0-RELEASE-NOTES.md [hunk 4]
```diff
@@ -62,4 +62,28 @@ methods: {
 }
 ```
 
-Please see [related code change](https://github.com/strongloop/loopback/pull/2174) here.
\ No newline at end of file
+Please see [related code change](https://github.com/strongloop/loopback/pull/2174) here.
+
+## remove unused user properties
+The following properties are removed from the common User model in 3.0:
```
Comment (bajtos) [inline 3.0-RELEASE-NOTES.md]: I think `built-in User model` is better than `common User model`.

**Comment by bajtos** [comment]
Two more comments, the PR looks good otherwise. No further review is necessary, please squash the commits before landing (and rebase on top of the current master).

**Pull request merged**